### PR TITLE
fix: #308 include req.readableAborted flag to defaultSuccessfulRequestMessageProvider

### LINKED ISSUE #308: Message "request completed" when request not completed
[Demo on codesandbox](https://codesandbox.io/p/sandbox/reverent-wiles-jnyyj5?file=%2Ftest.js%3A1%2C1&runonclick=1&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clp2sobhw00073b5wqnen9gpi%2522%252C%2522sizes%2522%253A%255B55.52397868561279%252C44.47602131438721%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clp2sobhw00033b5wwjudr9aj%2522%257D%252C%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clp2ssmri010i3b5wqiajs8lo%2522%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clp2sun0v01py3b5wwpij5uqm%2522%257D%252C%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clp5qk2oo00dc3b5wwtvpejm1%2522%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clp2sobhw00063b5wavlo2o8v%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B100%252C0%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clp2sobhw00033b5wwjudr9aj%2522%253A%257B%2522id%2522%253A%2522clp2sobhw00033b5wwjudr9aj%2522%252C%2522activeTabId%2522%253A%2522clp2ssldu00yh3b5wvnos5ds6%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Findex.js%2522%252C%2522id%2522%253A%2522clp2ssldu00yh3b5wvnos5ds6%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fpackage.json%2522%252C%2522id%2522%253A%2522clp2t9isd03kx3b5w34ilb7fn%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252F.codesandbox%252Ftasks.json%2522%252C%2522id%2522%253A%2522clp5q5skg00oc3b5wbvymhpgq%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%257D%252C%2522clp2sobhw00063b5wavlo2o8v%2522%253A%257B%2522id%2522%253A%2522clp2sobhw00063b5wavlo2o8v%2522%252C%2522tabs%2522%253A%255B%255D%257D%252C%2522clp2ssmri010i3b5wqiajs8lo%2522%253A%257B%2522id%2522%253A%2522clp2ssmri010i3b5wqiajs8lo%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Ftest.js%2522%252C%2522id%2522%253A%2522clp2stm2v01e93b5wv9dhal6u%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clp2stm2v01e93b5wv9dhal6u%2522%257D%252C%2522clp2sun0v01py3b5wwpij5uqm%2522%253A%257B%2522id%2522%253A%2522clp2sun0v01py3b5wwpij5uqm%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522start%2522%252C%2522id%2522%253A%2522clp2t3s5401d73b5wmrwjf4rl%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clp2t3s5401d73b5wmrwjf4rl%2522%257D%252C%2522clp5qk2oo00dc3b5wwtvpejm1%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clp5qjzww00d93b5wkiy3a2sq%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clp5qk2ty001iefibckgmbmyt%2522%257D%255D%252C%2522id%2522%253A%2522clp5qk2oo00dc3b5wwtvpejm1%2522%252C%2522activeTabId%2522%253A%2522clp5qjzww00d93b5wkiy3a2sq%2522%257D%257D%252C%2522showDevtools%2522%253Afalse%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

Request was aborted by `express.text` middleware because:
`Request body length does not match content-length header` (I know it's more like hypothetical situation, but anyway)

But there is `request completed` message from pino-http because `res.writableEnded` is `true`.
Maybe it is better to change condition for [defaultSuccessfulRequestMessageProvider](https://github.com/pinojs/pino-http/blob/44c61b76c0c05dace6b4c701d12c6a546c6ad013/logger.js#L262) to something like this:
```js
  return !req.readableAborted && res.writableEnded ? 'request completed' : 'request aborted'
```

### PR DESCRIPTION
**Pull request by someden**
Closes #308

### CODE REVIEW

**Review by mcollina — CHANGES_REQUESTED** [review]
Thanks for opening a PR! Can you please add a unit test?

**Comment by someden** [comment]
@mcollina done

**Comment by mcollina** [comment]
A test is failing on node v14

**Comment by someden** [comment]
@mcollina, do you have some plans to drop node v14 support?

**Comment by mcollina** [comment]
Of course, might actually be time to drop v14 and v16.

**Comment by mcollina** [comment]
Could you remove them from CI? I will bump the major.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -39,7 +39,7 @@ jobs:
       contents: read
     strategy:
       matrix:
-        node-version: [14, 16, 18]
+        node-version: [18]
```
Comment (mcollina) [inline .github/workflows/ci.yml:42]:
```suggestion
        node-version: [18, 20, 21]
```

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**